delft/sequenceLabelling/preprocess.py

```diff
@@ -385,6 +618,9 @@ def load(cls, file_path):
                     setattr(self, key, preprocessor)
                 else:
                     setattr(self, key, val)
+
+        # fix typing issue for integer keys which become string :/
+        self.indice_tag = {i: t for t, i in self.vocab_tag.items()}
```

Comment:
I'm checking the tests and one of the old tests is failing because the vocab_tag is None. 
I'm wondering if `vocab_tag` is ever not None. Since the self instance is created here and the `fit()` is not called. 
Maybe I'm missing something here?

Reply:
at this stage it's loading a saved Preprocessor object, so the vocab is always instaciated. 
We never save a preprocessor without vocab, so without fit called on it before

Reply:
I see. 

I'm now trying to fix the test by adding some data and calling the `fit()` method. However, the `self.indice_tag` is `None` after instantiating the preprocessor and calling the `fit()` method. 

Where this indice_tag is supposed to be populated? Shouldn't be done in the `fit()` method?

Reply:
In the previous version, `self.indice_tag` was always reconstructed when calling `inverse_transform()`, which was a waste! Now it's constructed from  `self.vocab_tag` only the first call to `inverse_transform()`:

https://github.com/kermitt2/delft/blob/dd748eec055ecb103a92ca4862d580f0be19cf10/delft/sequenceLabelling/preprocess.py#L550-L556

Reply:
Ok. Indeed it's better to cache the results. What about adding `        self.indice_tag = {i: t for t, i in self.vocab_tag.items()}` also at the end of the `fit()` method in Preprocessor?

Reply:
well the problem is that we need it in the `load()`, because dict with integer as key cannot be restored correctly by the json.load() - they will be transformed into strings.

Reply:
Yes. We leave it in the `load()` and we add it in the `fit()`.

Reply:
but otherwise if we have the `self.indice_tag = {i: t for t, i in self.vocab_tag.items()}` both at the end of `fit()` and `load()` I think this is no problem

Reply:
👍